rfs/fuse.go

```diff
@@ -0,0 +1,69 @@
+package rfs
+
+import (
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/Zamerykanizowana/replicated-file-system/config"
+	"github.com/Zamerykanizowana/replicated-file-system/logging"
+	"github.com/hanwen/go-fuse/v2/fs"
+	"github.com/hanwen/go-fuse/v2/fuse"
+	"go.uber.org/zap"
+)
+
+type RfsFuseServer struct {
+	LoopbackRoot *fs.LoopbackRoot
+	RfsRoot      fs.InodeEmbedder
+	Config       config.Config
+	Server       *fuse.Server
+}
+
+func (r *RfsFuseServer) Mount() error {
+	sec := time.Second
+
+	s, err := fs.Mount(r.Config.Paths.FuseDir, newRfsRoot(r.LoopbackRoot, nil, "", nil), &fs.Options{
+		AttrTimeout:  &sec,
+		EntryTimeout: &sec,
+	})
+
+	if err == nil {
+		r.Server = s
+	}
+
+	return err
+}
+
+func (r *RfsFuseServer) Wait() {
+	zap.L().Info(
+		"unmount by calling",
+		zap.String(
+			"cmd",
+			fmt.Sprintf("fusermount -u %s", r.Config.Paths.FuseDir),
```

Comment:
Good to know! I used `Sprintf` as I find that to be a bit more readable/elegant but it's just a personal preference. Didn't know that it uses `reflect` though.

Reply:
it has to since it accepts an array of `interfaces`, I've found people usually are afraid of concatenation for perf reasons (Java world), so just wanted to point that out in case you had that doubt ;)